MoltenVK/MoltenVK/Utility/MVKEnvironment.cpp

```diff
@@ -26,10 +26,9 @@
 // Return the expected size of MVKConfiguration, based on contents of MVKConfigMembers.def.
 static constexpr uint32_t getExpectedMVKConfigurationSize() {
 #define MVK_CONFIG_MEMBER(member, mbrType, name)         cfgSize += sizeof(mbrType);
-	uint32_t cfgSize = 0;
+	size_t cfgSize = 0;
 #include "MVKConfigMembers.def"
-	cfgSize += kMVKConfigurationInternalPaddingByteCount;
```

Comment:
@billhollings This seemed to be calculating the incorrect size, but perhaps that meant I added the config member incorrectly? I changed this function to align the expected size to the alignment of `MVKConfiguration`; please let me know if something is "off" here.

Reply:
Have a look at the description of `kMVKConfigurationInternalPaddingByteCount`. It's sole purpose is to adjust the calculation of:

`static_assert(getExpectedMVKConfigurationSize() == sizeof(MVKConfiguration), "MVKConfigMembers.def does not match the members of MVKConfiguration.");
`

And the purpose of **that** `static_assert()` is to ensure that the members of `MVKConfiguration` are all accounted for in `MVKConfigMembers.def` (ie- a dev mod isn't made to `MVKConfiguration` without adding the corrsponding entry in `MVKConfigMembers.def`), which would be tough to debug.

The reason `kMVKConfigurationInternalPaddingByteCount` is needed is to account for alignment padding that the compiler adds to `sizeof(MVKConfiguration)`.

So, the solution to the issue you have dealt with here is to adjust the value of `kMVKConfigurationInternalPaddingByteCount` accordingly. In this specific case, the value should be set to `0`, instead of removing the padding calculation. The next time an entry is added to `MVKConfiguration`, it will need to be set to a different value, and the padding calculation needs to be available.